File: schemas/multi-language-i18n/drizzle/locale_settings.ts
// locale_settings: Per-locale formatting and regional settings.
// See README.md for full design rationale.

import { pgTable, uuid, text, integer, timestamp } from "drizzle-orm/pg-core";
import { locales } from "./locales";

export const localeSettings = pgTable(
  "locale_settings",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    localeId: uuid("locale_id")
      .unique()
      .notNull()
      .references(() => locales.id, { onDelete: "cascade" }),
    dateFormat: text("date_format"),
    timeFormat: text("time_format"),
    numberFormat: text("number_format"),
    currencyCode: text("currency_code"),
    currencySymbol: text("currency_symbol"),
    firstDayOfWeek: integer("first_day_of_week").notNull().default(1),
    measurementSystem: text("measurement_system"),
    createdAt: timestamp("created_at", { withTimezone: true })
      .notNull()
      .defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true })
      .notNull()
      .defaultNow()
      .$onUpdateFn(() => new Date()),
  }
);

File: schemas/multi-language-i18n/spacetimedb/locale_settings.rs
// locale_settings: Regional formatting preferences per locale.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = locale_settings, public)]
pub struct LocaleSetting {
    #[primary_key]
    pub id: String, // UUID

    #[unique]
    pub locale_id: String, // FK → locales.id (cascade delete)

    pub date_format: Option<String>,
    pub time_format: Option<String>,
    pub number_format: Option<String>,
    pub currency_code: Option<String>,
    pub currency_symbol: Option<String>,
    pub first_day_of_week: i32,
    pub measurement_system: Option<String>,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
